src/DynamoCoreWpf/UI/GuidedTour/Guide.cs

```diff
@@ -106,12 +107,29 @@ public void Next(GuidedTourMovementEventArgs args)
             {
                 nextStep = (from step in GuideSteps where step.Sequence == args.StepSequence + 1 select step).FirstOrDefault();
                 if (nextStep != null)
+                {
                     nextStep.Show();
+
+                    //In case the UIAutomation info is set for the Step we execute the action when the Next button is pressed
+                    if (nextStep.UIAutomation != null)
+                    {
+                        ExecuteUIAutomationStep(nextStep.UIAutomation, true);
+                    }
+                }
+
             }
 
             CurrentStep = (from step in GuideSteps where step.Sequence == args.StepSequence select step).FirstOrDefault();
             if (CurrentStep != null)
+            {
                 CurrentStep.Hide();
+
+                //Disable the current action automation that is executed for the Current Step (if there is one)
+                if (CurrentStep.UIAutomation != null)
+                {
+                    ExecuteUIAutomationStep(CurrentStep.UIAutomation, false);
```

Comment:
can we remove the added code for nextStep and simply execute UIAutomation for each `current step`? I feel that is easier to understand and implement but not sure if there is other limitation

Reply:
the code added to NextStep() and PrevStep() were moved from Guide to Step class, so every time that the Step.Show() or Step.Hide() methods are called the UI Automation will be executed for hiding or showing the functionality.
see commit: https://github.com/DynamoDS/Dynamo/pull/11937/commits/65d76d3883ad2d050273a178016f454345b30ea8